Stellar-transaction.x

```diff
@@ -474,23 +474,87 @@ struct LiquidityPoolWithdrawOp
     int64 minAmountB; // minimum amount of second asset to withdraw
 };
 
-enum HostFunction
+enum HostFunctionType
 {
-    HOST_FN_INVOKE_CONTRACT = 0,
-    HOST_FN_CREATE_CONTRACT_WITH_ED25519 = 1,
-    HOST_FN_CREATE_CONTRACT_WITH_SOURCE_ACCOUNT = 2,
-    HOST_FN_CREATE_TOKEN_CONTRACT_WITH_SOURCE_ACCOUNT = 3,
-    HOST_FN_CREATE_TOKEN_CONTRACT_WITH_ASSET = 4
+    HOST_FUNCTION_TYPE_INVOKE_CONTRACT = 0,
+    HOST_FUNCTION_TYPE_CREATE_CONTRACT = 1,
+    HOST_FUNCTION_TYPE_INSTALL_CONTRACT_CODE = 2
+};
+
+enum ContractIDType
+{
+    CONTRACT_ID_FROM_PUBLIC_KEY = 0,
+    CONTRACT_ID_FROM_ASSET = 1
+};
+ 
+enum ContractIDPublicKeyType
+{
+    CONTRACT_ID_PUBLIC_KEY_SOURCE_ACCOUNT = 0,
+    CONTRACT_ID_PUBLIC_KEY_ED25519 = 1
+};
+
+struct InstallContractCodeArgs
+{
+    opaque code<SCVAL_LIMIT>;
+};
+
+enum CreateContractSourceType {
+    CONTRACT_SOURCE_REF = 0,
+    CONTRACT_SOURCE_INSTALLED = 1
+};
+
+union CreateContractSource switch (CreateContractSourceType type)
+{
+case CONTRACT_SOURCE_REF:
+    SCContractCode codeRef;
+case CONTRACT_SOURCE_INSTALLED:
```

Comment:
Should `CONTRACT_SOURCE_INSTALLED` be `CONTRACT_SOURCE_INSTALL` or something else? Installed makes it sounds like you’re using an already installed contract, which is actually the other option (`CONTRACT_SOURCE_REF`).

Reply:
Renamed, please let me know if it's still unclear.